model.py

```diff
@@ -298,30 +347,75 @@ def validate_short_name(self, key, value):
             )
         return value.upper()
 
+    @hybrid_property
+    def library_stage(self):
+        return self._library_stage
+
+    @library_stage.setter
+    def library_stage(self, value):
+        """A library can't unilaterally go from being in production to
+        not being in production.
+        """
+        if self.in_production and value != self.PRODUCTION_STAGE:
+            raise ValueError(
+                "This library is already in production; only the registry can take it out of production."
+            )
```

Comment:
What's the process for the registry to take it out of production? Bypass this setter and use `_library_stage`?

Reply:
The registry would set `registry_stage`, which doesn't require going through any code.

Reply:
Once registry_stage is no longer 'production', library_stage becomes unlocked and the library can set it to 'testing' or 'cancelled' if they want.

Since there's an out-of-band communication happening, the library can also leave library_stage alone and just coordinate with the library registry when the problem is fixed (assuming it's a problem that gets fixed).

Reply:
I guess what I'm saying is I don't think it's necessary to change `library_stage` to take a library out of production even if it was at the library's request.

Reply:
I could imagine a library wanting to be temporarily removed if they accidentally registered for production before they were ready, but it's probably unlikely, and you're right, they could coordinate with the registry admin.